.github/workflows/test-beta.yml

```diff
@@ -0,0 +1,25 @@
+name: test-beta
+
+on:
+  schedule:
+    - cron: "0 6 * * *"
+
+concurrency:
+  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: test
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@main
+      - name: Rustup
+        run: rustup install beta && rustup default beta
+      - uses: Swatinem/rust-cache@v2
+        with:
+          save-if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/next' }}
```

Comment:
github cache is quite space limited, and this only runs once a day so I don't think we should cache this.

```suggestion
```